### PR TITLE
Allow installing Python bindings into our Python Application

### PR DESCRIPTION
**Pull request by DanilaFe**
This PR adds a new target to our Makefile which builds on top of the `test-venv` and adds support for installing the Python bindings into the Python Application directory structure. This should make it possible to run scripts like `chplcheck` "out of the box", the same way we can run `chpldoc` (which requires Python for sphinx). The overarching goal is to be able to ship Python-based tools with Chapel, allowing our users to make use of `chplcheck` with no additional setup. It would also be nice if we could run the Python-based tools in CI jobs.

Reviewed by @jabraham17 and @mppf -- thanks!

## Testing

 - [x] c2chapel still works
 - [x] can invoke the linter from the command line without bindings installed

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -116,8 +116,13 @@ third-party-c2chapel-venv: FORCE
 	cd third-party && $(MAKE) c2chapel-venv; \
 	fi
 
+third-party-chapel-py-venv: frontend FORCE
+	cd third-party && $(MAKE) chapel-py-venv; \
```
Comment (jabraham17) [inline Makefile:120]: unnecessary backslash

**Review by jabraham17 — APPROVED** [review]
Looks good to me, but I am admittedly not very knowledgeable on the build system things

File: third-party/chpl-venv/Makefile [hunk 2]
```diff
@@ -92,6 +92,14 @@ chpldoc-venv: install-chpldeps
 
 c2chapel-venv: install-chpldeps
 
+chapel-py-venv: install-chpldeps
+	@# Install chapel-py's Python package into the virtual environment.
```
Comment (mppf) [inline third-party/chpl-venv/Makefile:96]: "virtual environment" -> "python app"

**Review by mppf — APPROVED** [review]
This looks fine to me but we need to stop calling it a virtual environment. There is a virtual environment used during builds but this is not it. What you are working with here is called a Python App. That's a pre-existing condition, but I'd ask that your PR not make it worse:
 * don't use "virtual environment" in the PR message or title
 * change the comment I noted

Also if you are able, it would be great to make a TODO issue about renaming directories and makefile targets to stop using `-venv` because this is a misnomer.

**Pull request merged**